### PR TITLE
#168977553 Build signup Endpoint using PostgreSQL

### PR DESCRIPTION
**Pull request by RusimbiP**
#### What does this PR do?
Have signup endpoint working
#### Description of Task to be completed?
Have the following endpoint working:
`POST api/v2/auth/signup`
#### How should this be manually tested?
After cloning this repo, cd into it and RUN **npm start** or npm run dev(to run in deveeloper mode)

- Using Postman test the endpoint above

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#168977553
#### Screenshots (if appropriate)
Screenshot (61)

#### Questions:
N/A

### CODE REVIEW

File: server/v2/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+    +-+-+-+-+-+-+-+-+-+ +-+-+ +-+-+-+-+-+-+ +-+-+-+-+-+ +-+-+-+-+ +-+-+-+ +-+-+-+-+ +-+-+-+-+-+
+  * |P|r|o|t|e|c|t|e|d| |b|y| |A|n|d|e|l|a| |H|o|n|o|r| |C|o|d|e| |3|r|d| |E|P|I|C| |V|a|l|u|e| *
```
Comment (Kasulejoseph) [inline server/v2/controllers/user.js]: I don't think these comments serve any purpose.. you can consider having this on the readme. Remove them from the codebase

File: server/v2/tests/user.test.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        });
+    });
+    it('Should return an error if a user tries to sign up without a firstname', (done) => {
+      chai
+        .request(app)
+        .post('/api/v2/auth/signup')
+        .send({
+          'lastname': 'PatricK',
+          'email': 'taken@teamwork.com',
+          'password': 'password',
```
[the remark this answers is not in the document]
Comment (RusimbiP) [reply server/v2/tests/user.test.js]: Ok

File: server/v2/controllers/user.js (hunk 1 above)
Comment (RusimbiP) [reply server/v2/controllers/user.js]: Alright.

File: server/v2/tests/user.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (RusimbiP) [reply server/v2/tests/user.test.js]: Changes made

File: server/v2/controllers/user.js (hunk 1 above)
Comment (RusimbiP) [reply server/v2/controllers/user.js]: Changes made

**Pull request merged**